src/chi-vue/src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,213 @@
+      this.animation = ThreeStepsAnimation.animationFactory(
+        () => {
+          this._animationClasses = transitioningClass;
+          this._backdropAnimationClasses = `${transitioningClass} ${closedClass}`;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = `${transitioningClass} ${activeClass}`;
+          this._backdropAnimationClasses = transitioningClass;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = activeClass;
+          this._backdropAnimationClasses = '';
+          this.$forceUpdate();
+          this.$emit(chiDrawerShownEvent);
+        },
+        ANIMATION_DURATION.MEDIUM
+      );
+    }
+
+    this.$emit(chiDrawerShowEvent);
+  }
+
+  hide() {
+    if (this.animation && !this.animation.isStopped()) {
+      this.animation.stop();
+    }
+
+    if (this._backdropAnimationClasses !== closedClass || this._animationClasses !== '') {
+      this.animation = ThreeStepsAnimation.animationFactory(
+        () => {
+          this._animationClasses = `${transitioningClass} ${activeClass}`;
+          this._backdropAnimationClasses = transitioningClass;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = transitioningClass;
+          this._backdropAnimationClasses = `${transitioningClass} ${closedClass}`;
+          this.$forceUpdate();
+        },
+        () => {
+          this._animationClasses = '';
+          this._backdropAnimationClasses = closedClass;
+          this.$forceUpdate();
+          this.$emit(chiDrawerHiddenEvent);
+        },
+        ANIMATION_DURATION.MEDIUM
+      );
+    }
+
+    this.$emit(chiDrawerHideEvent);
+  }
+
+  _documentClickHandler = (ev: Event): void => {
+    const drawerElement = this.$refs.drawerElement as HTMLElement;
+    const clickTarget = ev.target as HTMLElement;
+
+    if (!this.preventAutoHide) {
+      if (drawerElement.classList.contains(activeClass)) {
+          if (!contains(drawerElement, clickTarget)) {
+            this.hide();
+          }
+      }
+    }
+  };
+
+  async toggle() {
+    if (this.active) {
+      this.hide();
+    } else {
+      this.show();
+    }
+  }
+
+  beforeMount() {
+    this._animationClasses = this.active ? activeClass : '';
+    this._backdropAnimationClasses = this.active ? '' : closedClass;
+  }
+
+  beforeDestroy() {
+    document.removeEventListener('click', this.$data._documentClickHandler);
+  }
+
+  mounted() {
+    document.addEventListener('click', this.$data._documentClickHandler);
+  }
+
+  render() {
+    const closeButton = <button class={`
+      ${buttonClass}
+      -icon
+      -close`}
+      onClick={() => this.hide()}
+      aria-label="Close">
+      <div class={`${buttonContentClass}`}>
+        <i class={`${iconClass} icon-x`}></i>
+      </div>
+    </button>;
+
+    const drawer = (
+      <div
+        class={`
+        ${drawerClass}
```

Comment:
Just in case you have some classes in string variables and some others in array variables, you can use this method to join them: 
```JavaScript
class={[
  drawerClass,
  animatedClass, 
  ...this.animationClasses,
  etc
].filter(cl => cl).join(" ")}
```